test/spawn_future_test.cpp

```diff
@@ -0,0 +1,632 @@
+}
+
+struct identity_scope final {
+  template <typename Sender>
+  auto nest(Sender&& sender) noexcept(
+      std::is_nothrow_constructible_v<unifex::remove_cvref_t<Sender>, Sender>) {
+    return static_cast<Sender&&>(sender);
+  }
+};
+
+TEST(spawn_future_test, spawn_future_accepts_non_standard_scope_types) {
+  identity_scope idscope;
+  auto future = unifex::spawn_future(unifex::just(42), idscope);
+
+  auto ret = unifex::sync_wait(std::move(future));
+
+  ASSERT_TRUE(ret.has_value());
+  EXPECT_EQ(42, *ret);
+}
+
+TEST(spawn_future_test, spawn_future_is_pipeable) {
+  identity_scope idscope;
+  unifex::just() | unifex::spawn_future(idscope) | unifex::sync_wait();
+}
+
+TEST(spawn_future_test, spawn_future_accepts_allocators_of_non_bytes) {
+  identity_scope idscope;
+  unifex::spawn_future(unifex::just(), idscope, std::allocator<int>{}) |
+      unifex::sync_wait();
+}
+
+TEST(
+    spawn_future_test,
+    spawn_future_accepts_allocators_of_non_bytes_when_piped) {
+  identity_scope idscope;
+
+  unifex::just() | unifex::spawn_future(idscope, std::allocator<int>{}) |
+      unifex::sync_wait();
+}
+
+struct destruction_sensor final {
+  explicit destruction_sensor(int& constructs, int& destructs) noexcept
+    : constructs_(&constructs)
+    , destructs_(&destructs) {
+    ++(*constructs_);
+  }
+
+  destruction_sensor(const destruction_sensor& other) noexcept
+    : destruction_sensor(*other.constructs_, *other.destructs_) {}
+
+  ~destruction_sensor() { ++(*destructs_); }
+
+private:
+  int* constructs_;
+  int* destructs_;
+};
+
+TEST(
+    spawn_future_test,
+    operation_results_are_destroyed_when_future_is_immediately_discarded) {
+  unifex::v2::async_scope scope;
+
+  int valueConstructs = 0;
+  int valueDestructs = 0;
+  (void)unifex::spawn_future(
+      unifex::just(destruction_sensor{valueConstructs, valueDestructs}), scope);
+
+  int errorConstructs = 0;
+  int errorDestructs = 0;
+  (void)unifex::spawn_future(
+      // just_error is akward because future<> only supports exception_ptr
+      // errors so an error of type destruction_sensor requires hoop-jumping
+      unifex::just() | unifex::then([&]() {
+        throw destruction_sensor{errorConstructs, errorDestructs};
+      }),
+      scope);
+
+  unifex::sync_wait(scope.join());
+
+  EXPECT_EQ(valueConstructs, valueDestructs);
+  EXPECT_EQ(errorConstructs, errorDestructs);
+}
+
+TEST(spawn_future_test, happy_path_lacks_double_deletes) {
+  unifex::v2::async_scope scope;
+
+  int valueConstructs = 0;
+  int valueDestructs = 0;
+
+  unifex::spawn_future(
+      unifex::just(destruction_sensor{valueConstructs, valueDestructs}),
+      scope) |
+      unifex::sync_wait();
+
+  unifex::sync_wait(scope.join());
+
+  EXPECT_EQ(valueConstructs, valueDestructs);
+}
+
+TEST(spawn_future_test, spawn_future_of_never_sender_does_not_hang) {
+  unifex::v2::async_scope scope;
+  auto future = unifex::spawn_future(unifex::never_sender{}, scope);
+
+  unifex::sync_wait(unifex::when_all(
```

Comment:
same as below: Is there a way to assert that cancellation took effect?

Reply:
`never_sender` will never complete unless it receives a stop request.  If cancellation doesn't take effect, the spawned `never_sender` operation will never complete, leaving the scope open forever, leaving the `scope.join()` (and thus the whole test case) blocked forever.

In other words, if this test finishes, cancellation took effect; conversely, if cancellation doesn't take effect, the test will hang forever.

Does that satisfy or would you prefer a more affirmative assertion?